scraper/scraper_bot.py

```diff
@@ -134,141 +154,401 @@ def __init__(self, config: ScraperBotConfig, condition_fn: Callable, parse_fn: C
         self.limit = config.limit
         self.embed_images = config.embed_images
         self.hf_dataset_name = config.hf_dataset_name
-        self.parse_fn = parse_fn        
+        self.parse_fn = parse_fn
         self.condition_fn = condition_fn
-    
+
     @property
-    def url(self) -> str:
-        return f'{self.base_url}/channels/{self.channel_id}/messages?limit={self.limit}'
+    def schema(self):
+        schema = [f.name for f in fields(HFDatasetScheme)]
+        if not self.embed_images:
+            schema.remove("image")
+        return schema
 
+    @property
+    def url(self) -> str:
+        return f"{self.base_url}/channels/{self.channel_id}/messages?limit={self.limit}"
 
     @property
     def headers(self) -> Dict[str, str]:
-        if os.environ['IS_CI'] in 'true':
+        if os.environ["IS_CI"] in "true":
             return get_bot_headers()
         else:
             return get_user_headers()
 
+    @property
+    def fs_path(self) -> str:
+        path = f"datasets/{self.hf_dataset_name}/data"
+        return path
 
-    def _get_messages(self, after_message_id: str) -> List[Dict[str, Any]]:
+    @property
+    def repo_path(self) -> str:
+        return "data"
+
+    def _create_repo(self) -> None:
+        print(f"Repo did not exist, creating new repo {self.hf_dataset_name}")
+        create_repo(
+            self.hf_dataset_name,
+            repo_type="dataset",
+            token=os.environ["HF_TOKEN"],
+            exist_ok=True,
+        )
+        self._update_readme()
+
+    def _update_readme(self) -> None:
+        readme_template = f"dataset_readme_template.md"
+        addition = CommitOperationAdd(
+            path_in_repo=f"README.md",
+            path_or_fileobj=os.path.join(os.path.dirname(__file__), readme_template),
+        )
+        preupload_lfs_files(
+            repo_id=self.hf_dataset_name,
+            repo_type="dataset",
+            token=os.environ["HF_TOKEN"],
+            additions=[addition],
+        )
+
+        create_commit(
+            repo_id=self.hf_dataset_name,
+            repo_type="dataset",
+            commit_message="Update README.md",
+            token=os.environ["HF_TOKEN"],
+            operations=[addition],
+        )
+
+    def _get_chunk_names(self) -> None:
+        fs = HfFileSystem(token=os.environ["HF_TOKEN"], skip_instance_cache=True)
```

Comment:
It's better to define `fs = HfFileSystem(token=os.environ["HF_TOKEN"], skip_instance_cache=True)` only once globally. It will cache some stuff internally (in memory) so hopefully running `_get_chunk_names`,  ` _get_detailed_chunk_names` and `_append_chunk` might run faster in some cases.

(same for other `HfFileSystem` definitions below)

Reply:
> It will cache some stuff internally (in memory) so hopefully running _get_chunk_names,  _get_detailed_chunk_names and _append_chunk might run faster in some cases.

(actually `fsspec` does cache the instance IIRC)

Reply:
I was getting some issues with the cached instance not having the newest files, this seemed to solve the issue but maybe I'm not understanding what the true problem was.

Reply:
i.e. after committing a new file to the repo, it wasn't in `_get_chunk_names` until I redefined this.